### ISSUE TITLE
Microsoft.CodeAnalysis.Editor.UnitTests.Diagnostics.DiagnosticsSquiggleTaggerProviderTests.Test_TagSourceDiffer is flaky

### ISSUE THREAD

**Issue by jasonmalinowski** [dotnet/roslyn#28062]
Example failure: https://ci.dot.net/job/dotnet_roslyn/job/master/view/Official%20Builds/job/windows_debug_unit32/2431/

```
MESSAGE:
System.TimeoutException : Failed to clean up listeners in a timely manner.\r\n ReenqueueWorkItem WorkCoordinator.NormalPriorityProcessor.cs 359\r\n---- System.OperationCanceledException : The operation was canceled.
+++++++++++++++++++
STACK TRACE:
at Microsoft.CodeAnalysis.Test.Utilities.UseExportProviderAttribute.After(MethodInfo methodUnderTest) ----- Inner Stack Trace ----- at System.Threading.CancellationToken.ThrowOperationCanceledException() at System.Threading.CancellationToken.ThrowIfCancellationRequested() at Roslyn.Test.Utilities.TaskJoinExtensions.JoinUsingDispatcherNoResult(Task task, CancellationToken cancellationToken) at Roslyn.Test.Utilities.TaskJoinExtensions.JoinUsingDispatcher(Task task, CancellationToken cancellationToken) at Microsoft.CodeAnalysis.Test.Utilities.UseExportProviderAttribute.After(MethodInfo methodUnderTest)
```

@heejaechang and @jinujoseph can this get quickly looked at? Hit a few of my PRs and other partner teams too.

**Comment by jasonmalinowski** [comment]
@heejaechang Chatting with Sam, this is flagging this test as being bad -- something is starting an async operation and never cancelling it.

**Comment by heejaechang** [comment]
@sharwell hmmm.. okay. so all pending works must be disposed? is that basically what this wants? test itself is passing right?

**Comment by heejaechang** [comment]
basically, it looks like test doesn't care pending work items in the queue as long as it got things it needs processed. and it close solution crawler in shutdown mode (which is VS shutdown) basically doesn't care anything pending.

**Comment by 333fred** [comment]
Another one: https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_release_unit32_prtest/15541/testReport/junit/Microsoft.CodeAnalysis.Editor.UnitTests.Diagnostics/DiagnosticsSquiggleTaggerProviderTests/Test_TagSourceDiffer/
https://github.com/dotnet/roslyn/pull/28285

**Comment by tmat** [comment]
Another one: https://ci.dot.net/job/dotnet_roslyn/job/dev15.8.x/job/windows_debug_unit64_prtest/127/testReport/junit/Microsoft.CodeAnalysis.Editor.UnitTests.Diagnostics/DiagnosticsSquiggleTaggerProviderTests/Test_TagSourceDiffer/

**Comment by DustinCampbell** [comment]
Hey gang -- this is *still* flaky and failing in October. Is this going to get looked at? If not, it needs to be disabled to avoid randomizing the team.

**Comment by heejaechang** [comment]
alright, looks like another waiter not disposed thing?

**Comment by jaredpar** [comment]
another hit https://dnceng.visualstudio.com/public/_build/results?buildId=28686&view=logs

**Comment by CyrusNajmabadi** [comment]
Hit this again in https://dnceng.visualstudio.com/public/_build/results?buildId=46355&view=logs

**Comment by JoeRobich** [comment]
Hit this in #31231 [Build Logs](https://dnceng.visualstudio.com/public/_build/results?buildId=54026)

**Comment by JoeRobich** [comment]
Hit this in #31623 [Build Logs](https://dnceng.visualstudio.com/public/_build/results?buildId=54707)

**Comment by tmat** [comment]
One more: https://dnceng.visualstudio.com/public/_build/results?buildId=59318&view=ms.vss-test-web.test-result-details

**Comment by jaredpar** [comment]
We should consider disabling the test at this point. It's flakiness is above 3% now.

**Comment by jaredpar** [comment]
@sharwell I was looking at the list unfiltered. I didn't realize they mixed PR and non-PR branches in that way.

**Comment by mavasani** [comment]
Another failure: https://dnceng.visualstudio.com/public/_build/results?buildId=69044&view=ms.vss-test-web.test-result-details

**Comment by jaredpar** [comment]
@mavasani can you send a PR to disable this test? If not I can try later in the week (underwater at the moment).

**Comment by heejaechang** [comment]
@dotnet/roslyn-infrastructure where should one get dump for the failed test?

**Comment by sharwell** [comment]
@heejaechang Dumps are not collected for failed tests. You would need to reproduce the failure locally.
